server/src/main/java/org/apache/druid/server/log/FileRequestLogger.java

```diff
@@ -57,27 +57,34 @@ public class FileRequestLogger implements RequestLogger
   private final File baseDir;
   private final DateTimeFormatter filePattern;
   private final Duration durationToRetain;
+  private final Duration rollPeriod;
   private final Object lock = new Object();
 
-  private DateTime currentDay;
+  private DateTime currentPeriodStart;
   private OutputStreamWriter fileWriter;
 
   public FileRequestLogger(
       ObjectMapper objectMapper,
       ScheduledExecutorService exec,
       File baseDir,
       String filePattern,
-      Duration durationToRetain
+      Duration durationToRetain,
+      Duration rollPeriod
   )
   {
     this.exec = exec;
     this.objectMapper = objectMapper;
     this.baseDir = baseDir;
     this.filePattern = DateTimeFormat.forPattern(filePattern);
     this.durationToRetain = durationToRetain;
+    this.rollPeriod = rollPeriod;
```

Comment:
Should we initialize this to the default value if it is passed as null?

Reply:
This is being defaulted in `FileRequestLoggerProvider`  before the request logger object is created.